apps/communications/contacts/js/views/details.js

```diff
@@ -153,11 +154,25 @@ contacts.Details = (function() {
   };
 
   var showEditContact = function showEditContact() {
+    stopNFC();
     Contacts.showForm(true, contactData);
   };
 
   var setContact = function cd_setContact(currentContact) {
     contactData = currentContact;
+    startNFC(currentContact);
+  };
+
+  var startNFC = function(contact) {
+    if ('mozNfc' in navigator) {
+      NFC.startListening(contact);
+    }
+  };
+
+  var stopNFC = function() {
+    if ('mozNfc' in navigator) {
+      NFC.stopListening();
+    }
   };
```

Comment:
I think you dont need this wrapper here. `NFC.method` must be check inf mozNfc is available or not. This must be transparent for the code using NFC object.